stl/inc/filesystem

```diff
@@ -196,15 +196,21 @@ namespace filesystem {
     template <>
     inline constexpr bool _Is_EcharT<char32_t> = true;
 
+#ifdef __cpp_lib_concepts
+    // clang-format off
+    template <class _Ty>
+    concept _Is_Source_impl = input_iterator<_Ty> && _Is_EcharT<iter_value_t<_Ty>>;

```

Comment:
Is this conforming? Should http://eel.is/c++draft/fs.class.path#fs.path.req-1.3 be updated?

Reply:
> Is this conforming?

Not conforming enough, I'm afraid. I was going to let the difference between _`Cpp17Destructible`_ and `destructible` slide - the latter requires `noexcept` destruction, whereas the former allows potentially-throwing destructors that do not actually throw - but your comment made me double-check and I had forgotten that _`Cpp17InputIterator`_ doesn't require default construction which is certainly too big a stretch.

*sigh* I think WG21 will need to define some exposition-only "`meow_iterator || Cpp17MeowIterator` concepts to constrain "old" components so they admit both old and new iterators. I don't think we want to write "The template parameter `Iterator` shall either meet the _`Cpp17InputIterator`_ requirements or model `input_iterator`." a la [move.iter.requirements]/1 all over the library spec.

Reply:
Fancy fix replaced with simple conforming fix.